service/project.go

```diff
@@ -134,9 +152,10 @@ func (s *ProjectService) CreateEnvironment(ctx context.Context, c model.Environm
 func (s *ProjectService) GetEnvironment(ctx context.Context, projectID, envID, authUserID uuid.UUID) (model.Environment, error) {
 	// TODO add project member authorization
 
-	_, err := s.Get(ctx, projectID, authUserID)
-	if err != nil {
+	if exists, err := s.Exists(ctx, projectID, authUserID); err != nil {
```

Comment:
Doesn't read smoothly - we are in an `Environment`-related method, and query `Exists()` and it's about a project.

Would expect either `ProjectExists()` or maybe also move environment-related operations to a separate service, similar to what we did with `project_membership`.

Deserves a thought, mainly up to you, but from my side a simple argument, a simple rule of thumb - environment has a separate fully-fledged model with creation and update counterparts, so it's at least CRUD, which seems significant enough to indeed separate it.

Another argument is obviously the expectation how will they grow (project and environment domain).

Also, from a usage / reader perspective, again, there are now for example two methods in this service:
- `Create()` - creates project
- `CreateEnvironment()` - creates environment for a project

Doesn't feel smooth...

Reply:
Might be a separate PR of course

Reply:
I created an issue #23. We can discuss it there without blocking this PR. Also I am going to drop the second commit that ads `exists` method for now.